### ISSUE TITLE
Please include old security fixes

### ISSUE THREAD

**Issue by asarubbo** [mdadams/jasper#26]
Hello,

at gentoo we add the following patches that the maintainer found on the other distro:

https://dev.gentoo.org/~soap/distfiles/jasper-1.900.3_p20161014-patches.tar.bz2

Could you check and include what for you is suitable to go into upstream? thanks

**Comment by mdadams** [comment]
All of the patch files that you provided applied cleanly except for the following:
1) jasper-1.900.3-CVE-2008-3520.patch
2) jasper-1.900.3-libjasper-stepsizes-overflow.patch
For item 2, I was able to deduce the nature of the change and make a fix that I think is equivalent.
For item 1, it was a huge mess.  I think that every change was in conflict with the current code.  I quickly walked through the patch file for this item and it seems that all of the changes being made are making a change similar to what is already in the code base.  Namely, in order to avoid integer overflow when computing sizes of memory chunks to allocate with malloc and friends, some helper functions were added that internally do a safe integer multiply and check for overflow.  So, unless I missed anything in my manual scan of the patch file, all of bugs addressed by this patch file have already been dealt with.

So, in summary, unless I have missed anything from item 1, all of the patches that you have provided have now been applied to the code base (with some minor changes in some cases).  I have tagged a new release called version-1.900.5 with all of these changes included.  So, I am marking this issue as closed.
